horovod/_keras/__init__.py

```diff
@@ -73,8 +79,13 @@ def init_aggregation_vars():
                             elif isinstance(grad, tf.IndexedSlices):
                                 raise AssertionError(
                                     "IndexedSlices are not supported when `self._aggregation_frequency` > 1 and `self._sparse_as_dense is False")
+                            if self.grad_updated_sizes_dict:
+                                assert (idx, self.grad_updated_sizes_dict)
```

Comment:
I'm not entirely sure what this assert checks (i guess this is python2 style which may be before I was born), can we be more explicit? `assert idx in self.grad_updated_sizes_dict` or whatever it's supposed to check?

Reply:
This was just idiocy on my part, will fix.